src/error/error_checks.cpp

```diff
@@ -31,90 +31,90 @@ bool dimensionsTypecheck(const std::vector<IndexVar>& resultVars,
                          const IndexExpr& expr,
                          const std::vector<int>& dimensions) {
 
-  std::map<IndexVar,int> varSizes;
-  for (size_t i = 0; i < resultVars.size(); i++) {
-    IndexVar var = resultVars[i];
-    int dimension = dimensions[i];
-    if (util::contains(varSizes, var) && varSizes.at(var) != dimension) {
+  std::map<IndexVar,int> indexVarDimensions;
+  for (size_t mode = 0; mode < resultVars.size(); mode++) {
+    IndexVar var = resultVars[mode];
+    int dimension = dimensions[mode];
+    if (util::contains(indexVarDimensions, var) && indexVarDimensions.at(var) != dimension) {
       return false;
     }
     else {
-      varSizes.insert({var, dimension});
+      indexVarDimensions.insert({var, dimension});
     }
   }
 
   vector<const ReadNode*> readNodes = getReadNodes(expr);
   for (auto& readNode : readNodes) {
-    for (size_t i = 0; i < readNode->indexVars.size(); i++) {
-      IndexVar var = readNode->indexVars[i];
-      int dimension = readNode->tensor.getDimension(i);
-      if (util::contains(varSizes, var) && varSizes.at(var) != dimension) {
+    for (size_t mode = 0; mode < readNode->indexVars.size(); mode++) {
+      IndexVar var = readNode->indexVars[mode];
+      int dimension = readNode->tensor.getDimension(mode);
+      if (util::contains(indexVarDimensions, var) && indexVarDimensions.at(var) != dimension) {
         return false;
       }
       else {
-        varSizes.insert({var, dimension});
+        indexVarDimensions.insert({var, dimension});
       }
     }
   }
 
   return true;
 }
 
-static string addDimensionError(const IndexVar& var, int dim1, int dim2) {
+static string addDimensionError(const IndexVar& var, int dimension1, int dimension2) {
   return "Index variable " + util::toString(var) + " is used to index "
-         "dimensions of different sizes (" + util::toString(dim1) +
-         " and " + util::toString(dim2) + ").";
+         "modes of different dimensions (" + util::toString(dimension1) +
+         " and " + util::toString(dimension2) + ").";
 }
 
 std::string dimensionTypecheckErrors(const std::vector<IndexVar>& resultVars,
                                      const IndexExpr& expr,
                                      const std::vector<int>& dimensions) {
   vector<string> errors;
 
-  std::map<IndexVar,int> varSizes;
-  for (size_t i = 0; i < resultVars.size(); i++) {
-    IndexVar var = resultVars[i];
-    int dimension = dimensions[i];
-    if (util::contains(varSizes, var) && varSizes.at(var) != dimension) {
-      errors.push_back(addDimensionError(var, varSizes.at(var), dimension));
+  std::map<IndexVar,int> indexVarDimensions;
+  for (size_t mode = 0; mode < resultVars.size(); mode++) {
+    IndexVar var = resultVars[mode];
+    int dimension = dimensions[mode];
+    if (util::contains(indexVarDimensions, var) && indexVarDimensions.at(var) != dimension) {
```

Comment:
Code should not move over the 80th character column, because some of us have editors configured to be that wide. This is why we use shortened names some places. I recommend setting your editor to highlight this column.

Reply:
I'll make sure nothing goes over the 80th column.

Reply:
actually, this is proving to be quite difficult. Type `pcregrep -r --include '.*\.h$' --include '.*\.c$' --include '.*\.cpp$' '^.{81,}$' .` to see all the lines which go over the 80^th column.

Reply:
It's not the end of the world if you leave a few over 80 column now, but we should not add more in the future. Tricks I use include moving subexpressions out of larger expressions and shortening variable names. There's no automatic way for it.